drivers/storage/port/scsiport/scsiport.c

```diff
@@ -1113,67 +1160,75 @@ ScsiPortInitialize(
-                                               ResourceList,
-                                               FIELD_OFFSET(CM_RESOURCE_LIST,
-                                                List[0].PartialResourceList.PartialDescriptors) +
-                                                ResourceList->List[0].PartialResourceList.Count *
-                                                sizeof(CM_PARTIAL_RESOURCE_DESCRIPTOR),
-                                               FALSE,
-                                               &Conflict);
-                ExFreePool(ResourceList);
-
-                /* In case of a failure or a conflict, break */
-                if (Conflict || (!NT_SUCCESS(Status)))
-                {
-                    if (Conflict)
-                        Status = STATUS_CONFLICTING_ADDRESSES;
-                    break;
-                }
-            }
+        PDEVICE_OBJECT LowerPDO = NULL;
+
+        Status = IoReportDetectedDevice(DriverObject,
+                                        HwInitializationData->AdapterInterfaceType,
+                                        ConfigInfo.BusNumber,
+                                        PortConfig->SlotNumber,
+                                        ResourceList,
+                                        NULL,
+                                        TRUE,
+                                        &LowerPDO);
+
+        if (!NT_SUCCESS(Status))
+        {
+            DPRINT1("IoReportDetectedDevice failed. Status: 0x%x\n", Status);
+            __debugbreak();
+            break;
         }
 
-        /* Reset the Conflict var */
-        Conflict = FALSE;
+        DeviceExtension->LowerDeviceObject = IoAttachDeviceToDeviceStack(PortDeviceObject, LowerPDO);
+
+        ASSERT(DeviceExtension->LowerDeviceObject);
+
+        PortDeviceObject->Flags &= ~DO_DEVICE_INITIALIZING;
+
+        if (ResourceList)
+        {
+            ExFreePoolWithTag(ResourceList, TAG_SCSIPORT);
+        }
 
         /* Copy all stuff which we ever need from PortConfig to the DeviceExtension */
         if (PortConfig->MaximumNumberOfTargets > SCSI_MAXIMUM_TARGETS_PER_BUS)
             DeviceExtension->MaxTargedIds = SCSI_MAXIMUM_TARGETS_PER_BUS;
         else
             DeviceExtension->MaxTargedIds = PortConfig->MaximumNumberOfTargets;
 
-        DeviceExtension->BusNum = PortConfig->NumberOfBuses;
+        DeviceExtension->NumberOfBuses = PortConfig->NumberOfBuses;
         DeviceExtension->CachesData = PortConfig->CachesData;
         DeviceExtension->ReceiveEvent = PortConfig->ReceiveEvent;
         DeviceExtension->SupportsTaggedQueuing = PortConfig->TaggedQueuing;
         DeviceExtension->MultipleReqsPerLun = PortConfig->MultipleRequestPerLu;
 
+        /* Initialize bus scanning information */
+        size_t BusConfigSize = DeviceExtension->NumberOfBuses * sizeof(*DeviceExtension->Buses);
+        DeviceExtension->Buses = ExAllocatePoolZero(PagedPool, BusConfigSize, TAG_SCSIPORT);
```

Comment:
You're accessing this in `ScsiPortIoTimer` at DPC level, so this needs to be `NonPagedPool`.